### PR TITLE
CRUD for product

### PR DESCRIPTION
**Pull request by eeveesharp**

### CODE REVIEW

**Comment by hromau** [comment]
get rid folder .vs from git

**Comment by hromau** [comment]
there is a changes from BIN folder. u must have to delete it from Repo and from Git

**Comment by nantonov** [comment]
Update .gitignore and exclude unnecessary files

**Comment by nantonov** [comment]
Get rid of DAL and BL projects

**Review by nantonov — CHANGES_REQUESTED** [review]
Don't overcomplicate it

File: Shop/DAL/Interfaces/IProductServices.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DAL.Entities;
+
+namespace DAL.Interfaces
+{
+    public interface IProductServices
+    {
+        public void Remove(int id);
```
Comment (stas1408123) [inline Shop/DAL/Interfaces/IProductServices.cs:12]: Rename to delete

File: Shop/DAL/Interfaces/IProductServices.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+        public void Update(int id, Product product);
+
+        public void Create(Product product);
+
+        public Product Read(int id);
```
Comment (stas1408123) [inline Shop/DAL/Interfaces/IProductServices.cs:18]: Rename to GetProductById

File: Shop/DAL/Interfaces/IProductServices.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+using System.Threading.Tasks;
+using DAL.Entities;
+
+namespace DAL.Interfaces
+{
+    public interface IProductServices
+    {
+        public void Remove(int id);
+
+        public void Update(int id, Product product);
```
Comment (stas1408123) [inline Shop/DAL/Interfaces/IProductServices.cs:14]: You don't need id

File: Shop/DAL/Entities/Product.cs [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 namespace DAL.Entities
 {
-    internal class Product
+    public class Product
```
Comment (stas1408123) [inline Shop/DAL/Entities/Product.cs:9]: Rename ProductEntity

File: Shop/DAL/Implementations/ProductServices.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DAL.DB;
+using DAL.Entities;
+using DAL.Interfaces;
+
+namespace DAL.Implementations
+{
+    internal class ProductServices : IProductServices
```
Comment (stas1408123) [inline Shop/DAL/Implementations/ProductServices.cs:12]: Implement all methods

File: Shop/Shop/Controllers/ProductController.cs [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                db.Products.Remove(product);
+
+                db.SaveChanges();
+            }
+
+            return Ok();
+        }
+
+        [HttpPut]
+        public IActionResult Update()
```
Comment (stas1408123) [inline Shop/Shop/Controllers/ProductController.cs:46]: You should get input data in this method

File: Shop/Shop/Controllers/ProductController.cs [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            using (ApplicationContext db = new ApplicationContext())
+            {
+                Product product = db.Products.FirstOrDefault();
+
+                product.Price = 1;
```
Comment (stas1408123) [inline Shop/Shop/Controllers/ProductController.cs:52]: Remove

File: Shop/Shop/Controllers/ProductController.cs [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+                db.Products.Update(product);
+
+                db.SaveChanges();
+            }
+
+            return Ok();
+        }
+
+        [HttpPost]
+        public IActionResult Create()
```
Comment (stas1408123) [inline Shop/Shop/Controllers/ProductController.cs:63]: You should get input data in this method

File: Shop/Shop/Controllers/ProductController.cs [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            using (ApplicationContext db = new ApplicationContext())
+            {
+                Product phone = new Product { Name = "Phone", Price = 123, Quantity = 33 };
```
Comment (stas1408123) [inline Shop/Shop/Controllers/ProductController.cs:67]: remove

File: Shop/Shop/Controllers/ProductController.cs [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+            using (ApplicationContext db = new ApplicationContext())
+            {
+                resultProduct = db.Products.ToList();
+            }
+
+            return resultProduct; 
+        }
+
+        [HttpDelete]
+        public IActionResult Delete()
```
Comment (stas1408123) [inline Shop/Shop/Controllers/ProductController.cs:31]: You should get input data in this method

File: Shop/Shop/Product.cs [hunk 11]
```diff
@@ -0,0 +1,13 @@
+namespace Shop
+{
+    public class Product
```
Comment (stas1408123) [inline Shop/Shop/Product.cs:3]: Rename ProductViewModels
Add this class into folder which is called ViewModels

**Review by stas1408123 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**